### PR TITLE
Raw modules bundles

### LINKED ISSUE #1828: Should Modules have raw bundles?
Currently, for each bundle defined in `Algebra.Bundles`, there is a corresponding raw bundle, consisting of the operations but without the laws. The same is not true for `Algebra.Module.Bundles`. Should they be added?

### PR DESCRIPTION
**Pull request by Taneb**
Closes #1828

Turns out that the raw module bundles only need to be indexed on `Set` rather than `Ring` or `RawRing`!

### CODE REVIEW

File: src/Algebra/Module/Bundles.agda
Comment (jamesmckinna) [inline src/Algebra/Module/Bundles.agda:89]: I would love it if such 'substructure' definitions were available without having to be written out in full, cf. #2255 ... eg here, the `+ᴹ-rawMonoid` structure is already available, only the additional `_*ₗ_` field needs defining, but Agda makes such `record` re-combination a bit crunchier than I would like:
```agda
rawLeftSemimodule = record { M ; _*ₗ_ = _*ₗ_ } where module M = RawMonoid +ᴹ-rawMonoid
```
is a bit of a mouthful, but still more 'informative' than the 'flattened' version?

File: src/Algebra/Module/Bundles.agda
Comment (jamesmckinna) [inline src/Algebra/Module/Bundles.agda:130]: Ditto., et. seq.

File: src/Algebra/Module/Bundles.agda
Comment (jamesmckinna) [inline src/Algebra/Module/Bundles.agda:174]: Ditto.

File: src/Algebra/Module/Bundles.agda
Comment (jamesmckinna) [inline src/Algebra/Module/Bundles.agda:215]: Ditto.

File: src/Algebra/Module/Bundles.agda
Comment (jamesmckinna) [inline src/Algebra/Module/Bundles.agda:259]: Is this field not redundant?

File: src/Algebra/Module/Bundles.agda
Comment (jamesmckinna) [inline src/Algebra/Module/Bundles.agda:264]: See above... ditto.

File: src/Algebra/Module/Bundles.agda
Comment (jamesmckinna) [inline src/Algebra/Module/Bundles.agda:317]: ...

File: src/Algebra/Module/Bundles/Raw.agda
Comment (jamesmckinna) [inline src/Algebra/Module/Bundles/Raw.agda:43]: But this is unavoidable?

File: src/Algebra/Module/Bundles/Raw.agda
Comment (jamesmckinna) [inline src/Algebra/Module/Bundles/Raw.agda:69]: Etc.

**Review by jamesmckinna — APPROVED** [review]
This all looks great as-is, but if you can find a nice way to streamline the definition of the `Raw*` substructures, that would be great!

Similarly, long ago in advance of this issue being closed/a PR such as this being raised, I had left [a placeholder](https://github.com/agda/agda-stdlib/blob/bde655f8aa742ea82aa2c33e732353509d7dd397/src/Algebra/Module/Construct/Zero.agda#L40-L46) (commit https://github.com/agda/agda-stdlib/pull/1902/commits/5fbceb2151c566f7ad5f52ff7cda43f9c3f22543) in `Algebra.Module.Construct.Zero` for defining the `Raw*` substructures in-place for the `Zero` instance of `*Module`... 

... can this be tidied up now, or would you rather I did that after this is merged?

File: src/Algebra/Module/Bundles.agda
Comment (Taneb) [reply src/Algebra/Module/Bundles.agda:89]: Would that work in this case? `RawMonoid` has different field names to `RawLeftSemimodule`...

File: src/Algebra/Module/Bundles.agda
Comment (Taneb) [reply src/Algebra/Module/Bundles.agda:259]: Yes. My development process is write the type signature, write "name = ?", load file, press C-c C-r in the hole to get a record skeleton, fill in all the holes, delete the redundant fields. Sometimes I forget the last step.

File: src/Algebra/Module/Bundles.agda
Comment (jamesmckinna) [reply src/Algebra/Module/Bundles.agda:89]: Ach! I think you're right. Maddening! 
Having to add the `renaming` directives would vitiate the whole suggestion. Grrr.

File: src/Algebra/Module/Bundles/Raw.agda
Comment (jamesmckinna) [reply src/Algebra/Module/Bundles/Raw.agda:69]: Sigh..

File: src/Algebra/Module/Construct/Zero.agda
Comment (jamesmckinna) [inline src/Algebra/Module/Construct/Zero.agda:38]: Is this directive now also redundant?

Comment (Taneb) [reply src/Algebra/Module/Construct/Zero.agda:38]: I'm not sure it wasn't redundant to begin with, but either way I'll remove it

File: src/Algebra/Module/Construct/TensorUnit.agda
Comment (jamesmckinna) [inline src/Algebra/Module/Construct/TensorUnit.agda:72]: Any particular reason (here, and throughout above) to omit `R` as the domain of coefficients in the return type? Could that simplify the subsequent definition body (probably not!?)

Comment (Taneb) [reply src/Algebra/Module/Construct/TensorUnit.agda:72]: It's because the parameter isn't `R` but `R.Carrier`, and that's a bother to extract

**Comment by jamesmckinna** [comment]
Looking even better now. Many thanks for all the additional commits!
(And: still a shame about all the repetition. A possible refactoring for another time?)

File: src/Algebra/Module/Construct/Zero.agda
Comment (jamesmckinna) [reply src/Algebra/Module/Construct/Zero.agda]: (and I agree: it was redundant before! My mistake, then as now!)

File: src/Algebra/Module/Construct/TensorUnit.agda
Comment (jamesmckinna) [reply src/Algebra/Module/Construct/TensorUnit.agda:72]: D'oh! Apologies once again...

File: CHANGELOG.md
Comment (jamesmckinna) [inline CHANGELOG.md:43]: typo: `Construct`

Comment (Taneb) [reply CHANGELOG.md:43]: Good catch!

**Comment by jamesmckinna** [comment]
:tada: If there are no further comments by CoB tomorrow, I'll hit `merge`...

**Review by JacquesCarette — APPROVED** [review]
This is both great and massively screams for automation. No human should have had to type all of that.

**Comment by jamesmckinna** [comment]
Sorry for not merging when I had said I would. 
*very* glad that @JacquesCarette was able to offer a second pair of eyes, though`!
(and looking forward to the 'smart(er)' automation PR downstream... ;-))

**Pull request merged**
